### PR TITLE
YANG updates for -08 draft

### LINKED ISSUE #56: Remove optical-fiber-data grouping?
The grouping optical-fiber-data has been defined but it is not used in the model

Check whether it has been superseded by the grouping fiber-params or something needs to be added in grouping fiber-params and then remove it from the optical impairments topology YANG model

Co-authored-by: sergiobelotti <sergio.belotti@nokia.com>

### LINKED ISSUE #57: Remove optical-channel-data grouping?
The grouping optical-channel-data has been defined but it is not used in the model

Check whether it is needed and where or remove it from the model.

Co-authored-by: sergiobelotti <sergio.belotti@nokia.com>

### LINKED ISSUE #70: geolocation of amplifiers
Would it be possible to ad a geo localisation in amplifier elements

`uses tet:geolocation-container ;`

          +--ro (element)
             +--:(amplifier)
             |  +--ro amplifier
             |     +--ro geolocation
             |     |  +--ro altitude?    int64
             |     |  +--ro latitude?    geographic-coordinate-degree
             |     |  +--ro longitude?   geographic-coordinate-degree
             |     +--ro type-variety    string
             |     +--ro operational

...

### LINKED ISSUE #72: Is local list connectivity (LLC) per transponder or per transceiver?
As discussed during the 2021-05-25 weekly call: 
[tranceivers-llc-00.pptx](https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/files/6540921/tranceivers-llc-00.pptx)

### LINKED ISSUE #74: Issues in identifying a Link's and element's opposite direction Link and element
Current topology lists links as unidirectional Links. This means that a bidirectional OMS  A<--->B is modelled with two links
    Link1
A ---->B and
    Link2 
B ---->A  

In the case of parallel links between two nodes, we need to identify which links are associated
    Link1
A ---->B and
    Link2 
A ---->B and
    Link3 
B ---->A 
    Link4 
B ---->A

for example Links1/3 and Links2/4.
A solution to provide this association is to assume that the Link's src-tp-id and dest-tp-id refer to bidir termination-points 
         Link1
A tp1 ---->B and
         Link2 
A tp2---->B and
   Link3 
B ----> tp1 A 
   Link4 
B ----> tp2 A
However, RFC8345 does not state if termination-point is a unidir or a bidir object, so that two implementations are possible, and the use of src-tp-id or dst-tp-id to associate direction is not standard.

Secondly, the optical impairment topology is unidirectional, but often the tools that consume the topology need bidirectional topologies. In order to extract one bidir OMS out of the model, we would need to know which amplifiers, fibers are associated on the two OMS, especially if the number of elements is not exactly the same on both directions
For example:
A--->amp1 --->fiber1 ---> amp2  ---> fiber2(patch) --->amp3  --->fiber3 --->amp4-->B
A<---amp5<--- fiber4 <--- amp6 <--- fiber5 <--- amp7<---B

correct association: 
amp1 / amp5
fiber1 / fiber4
amp2+amp3 / amp6
fiber3 / fiber5
amp4 / amp7

For this purpose, would it be possible to add an attribute with a reference on link id/oms-element-id ? for example
(adding the reference of the link would solve the above tp bidir/unidir ambiguity)

       +--ro OMS-elements* [elt-index]
          +--ro elt-index                 uint16
          +--ro oms-element-uid?          string
          +--ro opposite-link-id    leafref to link-id
          +--ro opposite-element-id leafref to oms-element-uid
          +--ro (element)
             +--:(amplifier)
             |  +--ro amplifier
             |     +--ro type-variety    string
             |     +--ro operational
...
This would enable to build a bidir topo out of the optical impairment topo and adding the reference of the link would solve the above tp bidir/unidir ambiguity

### PR DESCRIPTION
**Pull request by italobusi**
Fix #72
Fix #74
Fix #70
Fix #56
Fix #57

Pending merge of ietf-ccamp-wg/ietf-ccamp-layer0-types-ext#29

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

### CODE REVIEW

File: ietf-optical-impairment-topology.tree [hunk 1]
```diff
@@ -143,8 +139,16 @@ module: ietf-optical-impairment-topology
        +--ro OMS-elements* [elt-index]
           +--ro elt-index                 uint16
           +--ro oms-element-uid?          string
+          +--ro reverse-element-ref
+          |  +--ro link-ref?
+          |  |       -> ../../../../../../../nt:link/link-id
+          |  +--ro oms-element-ref?   leafref
```
Comment (EstherLerouzic) [inline ietf-optical-impairment-topology.tree:145]: Shouldn't it be a list of references here ?

File: ietf-optical-impairment-topology.yang [hunk 2]
```diff
@@ -878,43 +725,66 @@ module ietf-optical-impairment-topology {
...
+            type leafref {
+              path "../../../../../../../nt:link/nt:link-id";
             }
-            case concentratedloss {
-              uses concentratedloss-params ;
+            description
+              "The reference to the OMS link which the OMS elements
+              belongs to.";
+          }
+          leaf oms-element-ref {
```
Comment (EstherLerouzic) [inline ietf-optical-impairment-topology.yang:741]: leaf list ?

File: ietf-optical-impairment-topology.tree [hunk 3]
```diff
@@ -34,7 +34,7 @@ module: ietf-optical-impairment-topology
     |     |        |     |       frequency-thz
     |     |        |     +--ro max-central-frequency?
     |     |        |     |       frequency-thz
-    |     |        |     +--ro minimum-channel-spacing?
```
[the remark this answers is not in the document]
Comment (sergiobelotti) [reply ietf-optical-impairment-topology.tree:37]: @EstherLerouzic : that was our understanding. If you look the comment above  what would be the parameter foreseen in the list of common-organizational-explicit-mode you will see that the expected attribute is the tunability in transmission.
So we added the parameter "central-frequency-step" . We think we made a mistake previously to put here instead the minimum-channel-spacing with the intention to represent "tunability".

File: ietf-optical-impairment-topology.tree [hunk 4]
```diff
@@ -256,6 +260,14 @@ module: ietf-optical-impairment-topology
             /tet:local-link-connectivities:
     +--ro add-path-impairments?    leafref
     +--ro drop-path-impairments?   leafref
+    +--ro transceiver* []
```
Comment (agva123) [inline ietf-optical-impairment-topology.tree:263]: Does this list need to be keyed with transponder-ref+transceiver-ref? If there is a scenario that one needs to look up the optical impairment of a specific transceiver then it perhaps needs a key.

File: ietf-optical-impairment-topology.tree [hunk 5]
```diff
@@ -183,10 +187,10 @@ module: ietf-optical-impairment-topology
                    +--ro loss    decimal64
   augment /nw:networks/nw:network/nw:node/tet:te
             /tet:tunnel-termination-point:
-    +--ro transceiver* []
-       +--ro transponder-ref?
+    +--ro transceiver* [transponder-ref transceiver-ref]
```
Comment (agva123) [inline ietf-optical-impairment-topology.tree:190]: Does this list need to be keyed with transponder-ref+transceiver-ref, or not?

Comment (sergiobelotti) [reply ietf-optical-impairment-topology.tree:190]: This is the piece of YANG tree have been presented by Italo and agreed during discussion in the weekly:

+--rw tunnel-termination-point* [tunnel-tp-id]
   +--rw tunnel-tp-id                           binary

OLD
   +--ro transceiver* []
NEW
   +--ro transceiver* [transponder-ref, tranceiver-ref]

   |  +--ro transponder-ref?
   |  |       -> ../../../../transponder/transponder-id
   |  +--ro tranceiver-ref?    leafref

**Pull request merged**